app/models/ability.rb

```diff
@@ -221,12 +223,24 @@ def find_by(id:)
       query_service.find_by(id: id)
     end
 
+    def find_proxy_file_sets(resource)
+      query_service.find_inverse_references_by(resource: resource, property: :proxied_file_id)
+    end
+
+    def proxy_parent_readable?(resource)
+      proxies = find_proxy_file_sets(resource)
+      return if proxies.empty?
+
+      proxy = proxies.first
```

Comment:
I don't think this will work if there are two ProxyFileSets in two playlists pointing to the same FileSet.